### PR TITLE
[SPARK-22191] [SQL] Add hive serde example with serde properties

### PR DESCRIPTION
**Pull request by crlalam**
## What changes were proposed in this pull request?

Added an example for, specifying serde with serde properties for hive tables using OPTIONS.

## How was this patch tested?

Verified manually by running examples with run-example.sh. Verified in Scala, java and python.

### CODE REVIEW

File: examples/src/main/java/org/apache/spark/examples/sql/hive/JavaSparkHiveExample.java [hunk 1]
```diff
@@ -124,6 +124,13 @@ public static void main(String[] args) {
     // ...
     // $example off:spark_hive$
 
+    // Hive serde's are also supported with serde properties.
+	String sqlQuery = "CREATE TABLE src_serde(key decimal(38,18), value int) USING hive"
```
Comment (dongjoon-hyun) [inline examples/src/main/java/org/apache/spark/examples/sql/hive/JavaSparkHiveExample.java]: Hi, @crlalam.
We use 2-space indentation in general.
FYI, maybe, you can see [Scala Coding Style](https://github.com/databricks/scala-style-guide).

**Comment by holdenk** [comment]
Jenkins OK to test

File: examples/src/main/java/org/apache/spark/examples/sql/hive/JavaSparkHiveExample.java [hunk 2]
```diff
@@ -124,6 +124,13 @@ public static void main(String[] args) {
     // ...
     // $example off:spark_hive$
 
+    // Hive serde's are also supported with serde properties.
+	String sqlQuery = "CREATE TABLE src_serde(key decimal(38,18), value int) USING hive" +
```
Comment (holdenk) [inline examples/src/main/java/org/apache/spark/examples/sql/hive/JavaSparkHiveExample.java]: The indentation here doesn't match the rest of the function.

**Comment by holdenk** [comment]
I love more examples, but is there a place we plan to put this in the documentation?

**Comment by HyukjinKwon** [comment]
ping @crlalam for @holdenk's comment.

**Pull request closed without merging**